Automattic/jetpack
# JS Components: Convert utility functions to TypeScript

#### Changes proposed in this Pull Request:
* This PR converts utility functions in `js-packages/components/tools` and `js-packages/components/lib` to TypeScript.
* It also converts their unit tests to TS

#### Other information:


#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:
* Checkout the PR
* Run `jetpack test js-packages/components js`
* Confirm that the tests pass
